packages/language-server/src/plugins/typescript/features/DiagnosticsProvider.ts

```diff
@@ -61,7 +80,8 @@ export class DiagnosticsProviderImpl implements DiagnosticsProvider {
             .filter(hasNoNegativeLines)
             .filter(isNoFalsePositive(document, tsDoc))
             .map(enhanceIfNecessary)
-            .map(swapDiagRangeStartEndIfNecessary);
+            .map(swapDiagRangeStartEndIfNecessary)
+            .filter(dedupDiagnostics());
```

Comment:
You said that some of the work results in duplicate diagnostics. If I understood correctly, only the diagnostics related to reactive statements are duplicated. If so, would it make more sense to add this filter only to `resolveNoopsInReactiveStatements` instead?

Reply:
- It is easier to filter after the conversion from ts.Diagnostic to VSCode Diagnostic, because the latter has a smaller API.
- By testing VSC Diagnostic (we could push to test all props of the type, not just those that are used in the conversion here), we're almost certain that a detected dup is absolutely the same thing. I considered exact dups were never desirable, so might as well put it at the end and be everything proof.

That said, I have no strong opinion on this. I can probably put something together to filter them out just of  `resolveNoopsInReactiveStatements`.

Your call?

Reply:
I prefer to have it inside `resolveNoopsInReactiveStatements` for two reasons:
1. Performance - probably doesn't really make a difference but that way only the ones that we know can be duplicates are filtered out
2. Colocation - reading this could I would assume that there are no duplicates because I would expect TS to know what it's doing and not create duplicate diagnostics on its own. Right now the new function is the only one creating new diagnostics out of existing ones so that is the only source for duplicates, so I prefer to have handled that there. Else someone could think "mhm this looks unnecessary" later on and remove it.

Reply:
OK I'm changing it.